src/librustc_ast/ast.rs

```diff
@@ -194,6 +194,9 @@ pub enum GenericArg {
     /// `'a` in `Foo<'a>`
     Lifetime(Lifetime),
     /// `Bar` in `Foo<Bar>`
+    ///
+    /// This is also used for const parameters
+    /// which are either `()` or a path.
```

Comment:
```suggestion
    /// This is also used for const parameters that are either `()` or a path,
    /// since we can't tell until type-checking (or sometimes name
    /// resolution), whether they refer to types or consts. This will be
    /// converted to consts later if we find out that's what they should have
    /// been all along.
```

Reply:
I removed this comment completely from this PR for now.

As I am currently working on https://github.com/rust-lang/rust/issues/67075 and https://github.com/rust-lang/rust/issues/66615 I will update the comment there. (Even if it will still take a few years until I get that done :laughing: )